### PR TITLE
test: introduce e2e tests for Contentful

### PR DESCRIPTION
**Pull request by axe312ger**
This adds a e2e test suite for Contentful.

Currently covering:

* rendering images via gatsby-image
* rendering images via gatsby-plugin-image
* the space used for this e2e test should cover all configuration options possible in Contentful right now

Later:
* we will add rendering for all content within the space. See #30854

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -409,6 +409,20 @@ jobs:
       - store_test_results:
           path: e2e-tests/visual-regression/cypress/results
 
+  e2e_tests_contentful:
+    <<: *e2e-executor
+    environment:
+    CYPRESS_PROJECT_ID: 2193cm
+    CYPRESS_RECORD_KEY: 57e9563e-af49-494c-837c-5af53c2d6f76
```
Comment (pieh) [inline .circleci/config.yml:416]: I think this is just whitespace indentation problem:
```suggestion
      CYPRESS_PROJECT_ID: 2193cm
      CYPRESS_RECORD_KEY: 57e9563e-af49-494c-837c-5af53c2d6f76
```

File: e2e-tests/contentful/cypress/integration/gatsby-image.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    .each(($el, i) => {
+      cy.wrap($el).should(`have.attr`, `srcset`)
+      cy.wrap($el).should(`have.attr`, `src`)
+      cy.wrap($el).matchImageSnapshot(`${type}-${i}`)
+    })
+}
+
+describe(`gatsby-image`, () => {
+  beforeEach(() => {
+    cy.visit("/gatsby-image")
```
Comment (LekoArts) [inline e2e-tests/contentful/cypress/integration/gatsby-image.js:42]: We normally use `.waitForRouteChange()` after `.visit()`:

https://github.com/gatsbyjs/gatsby/blob/07ae70d27a997c4db1689163581a9d8034d91200/e2e-tests/development-runtime/cypress/integration/navigation/linking.js#L3

It's inside `gatsby-cypress`:

https://github.com/gatsbyjs/gatsby/blob/07ae70d27a997c4db1689163581a9d8034d91200/e2e-tests/development-runtime/cypress/support/index.js#L1

File: e2e-tests/contentful/gatsby-config.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const path = require(`path`)
```
Comment (LekoArts) [inline e2e-tests/contentful/gatsby-config.js:1]: Unused `path`?

File: e2e-tests/contentful/gatsby-config.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = {
+  siteMetadata: {
+    title: `Gatsby Contentful e2e`,
+  },
+  plugins: [
+    {
+      resolve: `gatsby-source-contentful`,
+      options: {
+        spaceId: `k8iqpp6u0ior`,
+        accessToken: `hO_7N0bLaCJFbu5nL3QVekwNeB_TNtg6tOCB_9qzKUw`,
```
Comment (LekoArts) [inline e2e-tests/contentful/gatsby-config.js:12]: I assume this is read-only?

**Review by LekoArts — CHANGES_REQUESTED** [review]
Some smaller comments

File: e2e-tests/contentful/gatsby-config.js (hunk 4 above)
Comment (axe312ger) [reply e2e-tests/contentful/gatsby-config.js:12]: yes it is and a space only for this porpuse

File: e2e-tests/contentful/cypress/integration/gatsby-image.js (hunk 2 above)
Comment (axe312ger) [reply e2e-tests/contentful/cypress/integration/gatsby-image.js]: added

File: e2e-tests/contentful/gatsby-config.js (hunk 3 above)
Comment (axe312ger) [reply e2e-tests/contentful/gatsby-config.js]: removed

**Review by LekoArts — APPROVED** [review]
👍

**Pull request merged**